### ISSUE TITLE
System.Net.Security.Tests.ServerAsyncAuthenticateTest.ServerAsyncAuthenticate_EachServerUnsupportedProtocol_Fail fail in CI

### ISSUE THREAD

**Issue by Jiayili1** [dotnet/corefx#17083]
Configuration: OuterLoop_Fedora24_debug ([build#116](https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_fedora24_debug/116/testReport/))

Failed test: System.Net.Security.Tests.ServerAsyncAuthenticateTest.ServerAsyncAuthenticate_EachServerUnsupportedProtocol_Fail

Detail: https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_fedora24_debug/116/consoleText

Message:
~~~
Assert.Throws() Failure
        Expected: typeof(System.NotSupportedException)
        Actual:   typeof(System.NullReferenceException): Object reference not set to an instance of an object.
~~~

Stack Trace:
~~~
/mnt/resource/j/workspace/dotnet_corefx/master/outerloop_fedora24_debug/src/Common/src/System/Net/Security/Unix/SafeDeleteSslContext.cs(65,0): at System.Net.Security.SafeDeleteSslContext.Dispose(Boolean disposing)
           /mnt/resource/j/workspace/dotnet_corefx/master/outerloop_fedora24_debug/src/System.Net.Security/src/System/Net/Security/SecureChannel.cs(206,0): at System.Net.Security.SecureChannel.Close()
           /mnt/resource/j/workspace/dotnet_corefx/master/outerloop_fedora24_debug/src/System.Net.Security/src/System/Net/Security/SslState.cs(494,0): at System.Net.Security.SslState.Close()
           /mnt/resource/j/workspace/dotnet_corefx/master/outerloop_fedora24_debug/src/System.Net.Security/src/System/Net/Security/SslStream.cs(502,0): at System.Net.Security.SslStream.Dispose(Boolean disposing)
              at System.IO.Stream.Close()
           /mnt/resource/j/workspace/dotnet_corefx/master/outerloop_fedora24_debug/src/System.Net.Security/tests/FunctionalTests/ServerAsyncAuthenticateTest.cs(186,0): at System.Net.Security.Tests.ServerAsyncAuthenticateTest.<ServerAsyncSslHelper>d__11.MoveNext()
           --- End of stack trace from previous location where exception was thrown ---
              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
~~~

**Comment by karelz** [comment]
cc @Priya91 @ianhays @steveharter

**Comment by karelz** [comment]
Closed as dupe of #16516 (in the PR above).
